src/firestore/src/convert.rs

```diff
@@ -0,0 +1,227 @@
+// Copyright 2025 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+//! Contains integration tests for the generated `Convert` implementations.
+//!
+//! For generated code, we write hand-crafted integration tests *once* in the
+//! project. The generated implementations for `wkt::prost::Convert` usually
+//! reference a type generated by Protobuf, and these are `pub(crate)`.
+//! Therefore, the tests must be part of the crate that contains them, and
+//! cannot be a separate integration tests crate, or a file in the `tests/`
+//! directory.
+//!
+//! The strategy we follow is to write one test per category of generated code.
+//! Exhaustive tests of each generated code quickly get repetitive. The most
+//! sources of variation in these implementations are the type of fields. Such
+//! as:
+//!
+//! - Simple primitive fields
+//! - Enum fields
+//! - Repeated fields
+//! - Map fields
+//! - OneOf fields
+//! - Optional fields
+//!
+//! And then you need to consider variations on these, such as oneofs that
+//! contain enums, or maps that contain enums.
+
+#[cfg(test)]
+mod test {
+    use crate::generated::model;
+    use crate::google;
+    use wkt::prost::Convert;
+
+    #[test]
+    fn test_basic_fields() {
+        let sidekick = gtype::model::LatLng::new()
+            .set_latitude(12.5)
+            .set_longitude(34.5);
+        let proto = google::r#type::LatLng {
+            latitude: 12.5,
+            longitude: 34.5,
+        };
+
+        let got = sidekick.clone().cnv();
+        assert_eq!(got, proto);
+
+        let got = proto.clone().cnv();
+        assert_eq!(got, sidekick);
+    }
+
+    #[test]
+    fn test_enum_field() {
```

Comment:
thank you for naming the tests after the field types and not the message names used within.